### PR TITLE
Feature/relational db

### PR DESCRIPTION
**Pull request by zivgos**
Adding relational DB creation code and adding backend support for a sex attribute in the backend.

### CODE REVIEW

File: app/models/dog_document.py [hunk 1]
```diff
@@ -10,12 +8,15 @@
...
-    
+
 class DogType(str, Enum):
     FOUND: str = "found"
     LOST: str = "lost"
 
-    
+class DogSex(str, Enum):
+    MALE: str = "male"
+    LOST: str = "lost"
```
Comment (YanDav78) [inline app/models/dog_document.py]:
```suggestion
    FEMALE: str = "female"
```

File: app/databse.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+# Create a base class for declarative models
+Base = declarative_base()
+
+
+class Dog(Base):
+    __tablename__ = 'dogs'
+
+    # Dog data
+    ## Entry data
+    id = Column(ID_COLUMN_TYPE, primary_key=True, default=ID_CREATOR)
```
Comment (YanDav78) [inline app/databse.py:49]: Can we just use autoincrement primary key here?

File: app/databse.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+def get_connection_string(user: str, password: str, host: str, port: int = 5432, db: str = 'dogfinder',
+                          protocol: str = 'postgresql') -> str:
+    if protocol == 'sqlite':
+        return f"sqlite:///{db}.db"
+    else:
+        return f'{protocol}://{user}:{password}@{host}:{port}/{db}'
+
+
+def main():
```
Comment (YanDav78) [inline app/databse.py]: This method needs to be commented out?

File: app/databse.py (hunk 2 above)
Comment (zivgos) [reply app/databse.py:49]: That was my original choice, Tslil asked for UUIDs

Comment (YanDav78) [reply app/databse.py:49]: Why?

File: app/databse.py (hunk 3 above)
Comment (zivgos) [reply app/databse.py]: This is creating the connection string:
- The "if" clause runs when there is not DB data in the environment variables, so the protocol defaults to "sqlite".
- if the environment variables contain real DB data the "else" clause runs.

File: app/databse.py (hunk 2 above)
Comment (zivgos) [reply app/databse.py:49]: From what I remember she said that:
1. They are unique over all the tables.
2. They don't indicate how many records we have.
I can't tag her here, so maybe you can take it up with her and you can see what you prefer.

Comment (YanDav78) [reply app/databse.py:49]: I'll talk to her. Thank you!

File: app/databse.py (hunk 3 above)
Comment (YanDav78) [reply app/databse.py]: I meant the main method

Comment (zivgos) [reply app/databse.py]: Wasn't sure about this. Will comment out.
Thanks.

**Comment by zivgos** [comment]
Have more changes in the next PR

**Pull request closed without merging**